### PR TITLE
Replace all median grades and variances by standard deviation

### LINKED ISSUE #558: replace median by variance
in the export, result overview and course detail pages, we show a total median rating for the course. we realized that we don't know what this should actually mean, but that an average over all variances would be helpful. so these three places, the median should be replaced by the variance.

### PR DESCRIPTION
**Pull request by karyon**
this PR replaces #570, and still fixes #558.

it also replaces all median grades and variances by the standard deviation.

### CODE REVIEW

File: evap/evaluation/tools.py [hunk 1]
```diff
@@ -195,27 +194,27 @@ def _calculate_results_impl(course):
...
 
-def calculate_average_and_medium_grades(course):
-    """Determines the final average and median grades for a course."""
+def calculate_average_grades_and_deviation(course):
+    """Determines the final average grade and deviation for a course."""
     avg_generic_likert = []
     avg_contribution_likert = []
-    med_generic_likert = []
-    med_contribution_likert = []
+    var_generic_likert = []
```
Comment (janno42) [inline evap/evaluation/tools.py]: shouldn't this be ```dev_generic_likert```?
same for three more occurrences below.

File: evap/results/exporters.py [hunk 2]
```diff
@@ -13,54 +13,59 @@ class ExcelExporter(object):
...
+            'course_unfinished': xlwt.easyxf('alignment: horiz centre, wrap on, rota 90; borders: left medium, top medium; font: italic on'),
+            'total_voters': xlwt.easyxf('alignment: horiz centre; borders: left medium, bottom medium, right medium'),
+            'bold':          xlwt.easyxf('font: bold on'),
+            'border_left':   xlwt.easyxf('borders: left medium'),
+            'border_right':  xlwt.easyxf('borders: right medium'),
+            'border_top_bottom_right': xlwt.easyxf('borders: top medium, bottom medium, right medium')}
+
+        CUSTOM_COLOR_START = 8
+        NUM_GRADE_COLORS = 21 # 1.0 to 5.0 in 0.2 steps
+        NUM_DEVIATION_COLORS = 30 # about to change in next commit
```
Comment (janno42) [inline evap/results/exporters.py]: ?

File: evap/results/exporters.py [hunk 3]
```diff
@@ -13,54 +13,59 @@ class ExcelExporter(object):
...
+        self.STEP = 0.2 # we only have a limited number of custom colors
+        self.STEP_INVERSE = (0.2) # we only have a limited number of custom colors
```
Comment (janno42) [inline evap/results/exporters.py]: do we need that?

File: evap/results/templates/results_course_detail.html [hunk 4]
```diff
@@ -34,7 +34,7 @@
                         <th class="col-sm-2">{% trans "Type" %}</th>
                         <th class="col-sm-3">{% trans "Responsible" %}</th>
                         <th class="col-sm-2">{% trans "Average Grade" %}</th>
-                        <th class="col-sm-2">{% trans "Median Grade" %}</th>
+                        <th class="col-sm-2">{% trans "deviation" %}</th>
```
Comment (janno42) [inline evap/results/templates/results_course_detail.html]: Please capitalize like the others

**Comment by janno42** [comment]
in the excel export 0,0 deviations are missing (like in the middle below):
missing

File: evap/results/templates/results_semester_detail.html [hunk 5]
```diff
@@ -25,7 +25,7 @@
                     <th class="col-sm-1">{% trans "Type" %}</th>
                     <th class="col-sm-3">{% trans "Responsible" %}</th>
                     <th class="col-sm-1">{% trans "Average Grade" %}</th>
-                    <th class="col-sm-1">{% trans "Median Grade" %}</th>
+                    <th class="col-sm-1">{% trans "Deviation" %}</th>
```
Comment (janno42) [inline evap/results/templates/results_semester_detail.html]: please use "Standard deviation"

File: evap/results/exporters.py [hunk 6]
```diff
@@ -141,17 +146,17 @@ def export(self, response, ignore_not_enough_answers=False):
...
-            avg, med = calculate_average_and_medium_grades(course)
+            avg, var = calculate_average_grades_and_deviation(course)
             if avg:
-                writec(self, avg, ExcelExporter.grade_to_style(avg), cols=2)
+                writec(self, avg, self.grade_to_style(avg), cols=2)
             else:
                 self.write_two_empty_cells_with_borders()
 
-        writen(self, _("Overall Median Grade"), "bold")
+        writen(self, _("Overall Average Deviation"), "bold")
```
Comment (janno42) [inline evap/results/exporters.py]: "Overall Average Standard Deviation"

**Comment by janno42** [comment]
please remove the "Standard" in the excel export from the caption titles (evap/results/exporters.py, line 107), then :shipit:

**Comment by karyon** [comment]
i think this should be ready now, please have a look again

**Comment by janno42** [comment]
Looks ready to go! :+1:

**Pull request merged**